src/backend/distributed/utils/metadata_cache.c

```diff
@@ -294,6 +353,54 @@ LookupDistTableCacheEntry(Oid relationId)
 
 
 /*
+ * CompareShardIntervals acts as a helper function to compare two shard intervals
+ * by their minimum values, using the value's type comparison function.
+ *
+ * If a shard interval does not have min/max value, it's treated as being greater
+ * than the other.
+ */
+static int
+CompareShardIntervals(const void *leftElement, const void *rightElement,
+					  FmgrInfo *typeCompareFunction)
+{
+	ShardInterval leftShardInterval = *((ShardInterval *) leftElement);
```

Comment:
In my fast shard pruning work, I replaced this to pointer. But, since we've already tested this and get into a mini `code-freeze` we didn't want to rebase it in this PR.